astroquery/hsa/core.py

```diff
@@ -0,0 +1,364 @@
+
+        if verbose:
+            log.info(link)
+
+        response = self._request('HEAD', link, save=False, cache=cache)
+        response.raise_for_status()
+        local_filepath = self._request('GET', link, cache=True, save=True)
+
+        original_filename = re.findall('filename="(.+)"',
+                                       response.headers["Content-Disposition"])[0]
+        _, ext = os.path.splitext(original_filename)
+        if filename is None:
+            filename = observation_id
+
+        filename += ext
+
+        shutil.move(local_filepath, filename)
+
+        if verbose:
+            log.info("Wrote {0} to {1}".format(link, filename))
+
+        return filename
+
+    def query_hsa_tap(self, query, *, output_file=None,
+                      output_format="votable", verbose=False):
+        """
+        Launches a synchronous job to query HSA Tabular Access Protocol (TAP) Service
+
+        Parameters
+        ----------
+        query : string, mandatory
+            query (adql) to be executed
+        output_file : string, optional, default None
+            file name where the results are saved if dumpToFile is True.
+            If this parameter is not provided, the jobid is used instead
+        output_format : string, optional, default 'votable'
+            values 'votable' or 'csv'
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A table object
+        """
+        job = self._tap.launch_job(query=query, output_file=output_file,
+                                   output_format=output_format,
+                                   verbose=verbose,
+                                   dump_to_file=output_file is not None)
+        table = job.get_results()
+        return table
+
+    def get_tables(self, *, only_names=True, verbose=False):
+        """
+        Get the available table in HSA TAP service
+
+        Parameters
+        ----------
+        only_names : bool, optional, default True
+            True to load table names only
+        verbose : bool, optional, default False
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of tables
+        """
+        tables = self._tap.load_tables(verbose=verbose)
+        if only_names:
+            return [t.name for t in tables]
+        else:
+            return tables
+
+    def get_columns(self, table_name, *, only_names=True, verbose=False):
+        """
+        Get the available columns for a table in HSA TAP service
+
+        Parameters
+        ----------
+        table_name : string, mandatory
+            table name of which, columns will be returned
+        only_names : bool, optional, default True
+            True to load column names only
+        verbose : bool, optional, default False
+
+            flag to display information about the process
+        Returns
```

Comment:
Empty line is needed to get this rendered properly
```suggestion

        Returns
```

Reply:
👍